File: app/src/main/java/com/progix/fridgex/light/data/SharedPreferencesAccess.java
package com.progix.fridgex.light.data;

import android.content.Context;
import android.content.SharedPreferences;

public final class SharedPreferencesAccess {

    public static void saveBoolean(Context context, String key, Boolean value) {
        SharedPreferences sharedPreferences = context.getSharedPreferences("fridgex", Context.MODE_PRIVATE);
        SharedPreferences.Editor editor = sharedPreferences.edit();
        editor.putBoolean(key, value);
        editor.apply();
    }

    public static Boolean loadBoolean(Context context, String key) {
        return context.getSharedPreferences("fridgex", Context.MODE_PRIVATE).getBoolean(key, true);
    }

    public static void saveString(Context context, String key, String value) {
        SharedPreferences sharedPreferences = context.getSharedPreferences("fridgex", Context.MODE_PRIVATE);
        SharedPreferences.Editor editor = sharedPreferences.edit();
        editor.putString(key, value);
        editor.apply();
    }

    public static String loadString(Context context, String key) {
        return context.getSharedPreferences("fridgex", Context.MODE_PRIVATE).getString(key, "0");
    }

    public static Integer loadNightMode(Context context) {
        return context.getSharedPreferences("fridgex", Context.MODE_PRIVATE).getInt("mode", 2);
    }

    public static void saveNightMode(Context context, Integer value) {
        SharedPreferences sharedPreferences = context.getSharedPreferences("fridgex", Context.MODE_PRIVATE);
        SharedPreferences.Editor editor = sharedPreferences.edit();
        editor.putInt("mode", value);
        editor.apply();
    }

    public static Integer loadCartMode(Context context) {
        return context.getSharedPreferences("fridgex", Context.MODE_PRIVATE).getInt("cartMode", 1);
    }

    public static void saveCartMode(Context context, Integer value) {
        SharedPreferences sharedPreferences = context.getSharedPreferences("fridgex", Context.MODE_PRIVATE);
        SharedPreferences.Editor editor = sharedPreferences.edit();
        editor.putInt("cartMode", value);
        editor.apply();
    }

    public static void saveDate(Context context, Integer value) {
        SharedPreferences sharedPreferences = context.getSharedPreferences("fridgex", Context.MODE_PRIVATE);
        SharedPreferences.Editor editor = sharedPreferences.edit();
        editor.putInt("date", value);
        editor.apply();
    }

    public static Integer loadDate(Context context) {
        return context.getSharedPreferences("fridgex", Context.MODE_PRIVATE).getInt("date", 0);
    }

    public static void saveDailyRecipe(Context context, String key, String value) {
        SharedPreferences sharedPreferences = context.getSharedPreferences("fridgex", Context.MODE_PRIVATE);
        SharedPreferences.Editor editor = sharedPreferences.edit();
        editor.putString(key, value);
        editor.apply();
    }

    public static String loadDailyRecipe(Context context, String key) {
        return context.getSharedPreferences("fridgex", Context.MODE_PRIVATE).getString(key, "");
    }

    public static void saveFirstStart(Context context, Boolean value) {
        SharedPreferences sharedPreferences = context.getSharedPreferences("fridgex", Context.MODE_PRIVATE);
        SharedPreferences.Editor editor = sharedPreferences.edit();
        editor.putBoolean("firstStart", value);
        editor.apply();
    }

    public static Boolean loadFirstStart(Context context) {
        return context.getSharedPreferences("fridgex", Context.MODE_PRIVATE).getBoolean("firstStart", true);
    }

    public static void saveTheme(Context context, String value) {
        SharedPreferences sharedPreferences = context.getSharedPreferences("fridgex", Context.MODE_PRIVATE);
        SharedPreferences.Editor editor = sharedPreferences.edit();
        editor.putString("theme", value);
        editor.apply();
    }

    public static String loadTheme(Context context) {
        return context.getSharedPreferences("fridgex", Context.MODE_PRIVATE).getString("theme", "def");
    }

    public static void saveFont(Context context, Float value) {
        SharedPreferences sharedPreferences = context.getSharedPreferences("fridgex", Context.MODE_PRIVATE);
        SharedPreferences.Editor editor = sharedPreferences.edit();
        editor.putFloat("font", value);
        editor.apply();
    }

    public static Float loadFont(Context context) {
        return context.getSharedPreferences("fridgex", Context.MODE_PRIVATE).getFloat("font", 1.0f);
    }
}

File: app/src/main/java/com/progix/fridgex/light/data/SharedPreferencesAccess.kt
package com.progix.fridgex.light.data

import android.content.Context

object SharedPreferencesAccess {

    fun saveBoolean(context: Context, key: String?, value: Boolean) {
        val sharedPreferences = context.getSharedPreferences("fridgex", Context.MODE_PRIVATE)
        val editor = sharedPreferences.edit()
        editor.putBoolean(key, value)
        editor.apply()
    }

    fun loadBoolean(context: Context, key: String?): Boolean {
        val sharedPreferences = context.getSharedPreferences("fridgex", Context.MODE_PRIVATE)
        return sharedPreferences.getBoolean(key, true)
    }

    fun saveString(context: Context, key: String, value: String) {
        val sharedPreferences = context.getSharedPreferences("fridgex", Context.MODE_PRIVATE)
        val editor = sharedPreferences.edit()
        editor.putString(key, value)
        editor.apply()
    }

    fun loadString(context: Context, key: String): String? {
        val sharedPreferences = context.getSharedPreferences("fridgex", Context.MODE_PRIVATE)
        return sharedPreferences.getString(key, "0")
    }

    fun loadNightMode(context: Context): Int {
        val sharedPreferences = context.getSharedPreferences("fridgex", Context.MODE_PRIVATE)
        return sharedPreferences.getInt("mode", 2)
    }

    fun saveNightMode(context: Context, value: Int) {
        val sharedPreferences = context.getSharedPreferences("fridgex", Context.MODE_PRIVATE)
        val editor = sharedPreferences?.edit()
        editor?.putInt("mode", value)
        editor?.apply()
    }

    fun loadCartMode(context: Context): Int {
        val sharedPreferences = context.getSharedPreferences("fridgex", Context.MODE_PRIVATE)
        return sharedPreferences.getInt("cartMode", 1)
    }

    fun saveCartMode(context: Context, value: Int) {
        val sharedPreferences = context.getSharedPreferences("fridgex", Context.MODE_PRIVATE)
        val editor = sharedPreferences?.edit()
        editor?.putInt("cartMode", value)
        editor?.apply()
    }

    fun saveDate(context: Context, value: Int) {
        val sharedPreferences = context.getSharedPreferences("fridgex", Context.MODE_PRIVATE)
        val editor = sharedPreferences?.edit()
        editor?.putInt("date", value)
        editor?.apply()
    }

    fun loadDate(context: Context): Int {
        val sharedPreferences = context.getSharedPreferences("fridgex", Context.MODE_PRIVATE)
        return sharedPreferences.getInt("date", 0)
    }

    fun saveDailyRecipe(context: Context, key: String, value: String) {
        val sharedPreferences = context.getSharedPreferences("fridgex", Context.MODE_PRIVATE)
        val editor = sharedPreferences.edit()
        editor.putString(key, value)
        editor.apply()
    }

    fun loadDailyRecipe(context: Context, key: String): String? {
        val sharedPreferences = context.getSharedPreferences("fridgex", Context.MODE_PRIVATE)
        return sharedPreferences.getString(key, "")
    }

    fun saveFirstStart(context: Context, value: Boolean) {
        val sharedPreferences = context.getSharedPreferences("fridgex", Context.MODE_PRIVATE)
        val editor = sharedPreferences.edit()
        editor.putBoolean("firstStart", value)
        editor.apply()
    }

    fun loadFirstStart(context: Context): Boolean {
        val sharedPreferences = context.getSharedPreferences("fridgex", Context.MODE_PRIVATE)
        return sharedPreferences.getBoolean("firstStart", true)
    }

    fun saveTheme(context: Context, value: String) {
        val sharedPreferences = context.getSharedPreferences("fridgex", Context.MODE_PRIVATE)
        val editor = sharedPreferences.edit()
        editor.putString("theme", value)
        editor.apply()
    }

    fun loadTheme(context: Context): String? {
        val sharedPreferences = context.getSharedPreferences("fridgex", Context.MODE_PRIVATE)
        return sharedPreferences.getString("theme", "def")
    }

    fun saveFont(context: Context, value: Float) {
        val sharedPreferences = context.getSharedPreferences("fridgex", Context.MODE_PRIVATE)
        val editor = sharedPreferences.edit()
        editor.putFloat("font", value)
        editor.apply()
    }

    fun loadFont(context: Context): Float {
        val sharedPreferences = context.getSharedPreferences("fridgex", Context.MODE_PRIVATE)
        return sharedPreferences.getFloat("font", 1.0f)
    }
}